modules/sugar/src/main/ts/ephox/sugar/api/selection/WindowSelection.ts

```diff
@@ -43,7 +43,7 @@ const setRangeFromRelative = (win: Window, relative: Selection): void =>
       // If this selection is backwards, then we need to use extend.
       if (selection.setBaseAndExtent) {
         selection.setBaseAndExtent(start.dom(), soffset, finish.dom(), foffset);
-      } else if (selection.extend) {
+      } else if ((selection as any).extend) {
```

Comment:
I'm surprised, I would have thought we'd have a type for window selection? Or is it missing this method?

Reply:
That's not the problem here. It's that the window selection types don't include the `extend` property, as it's experimental: https://developer.mozilla.org/en-US/docs/Web/API/Selection/extend

Reply:
Hmm. It's in the TS types though? https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L14839

Reply:
Sorry I have that backwards looking further... the problem is the types say it always exists, but it doesn't exist on IE.

Reply:
it's one of those "experimental but everybody except IE11 has it, and it's absolutely essential" experimental technologies

Reply:
it's weird that this used to work, and upgrading dom-globals broke it, I guess it was removed from the interfaces? The history is too complicated for github to render so I can't easily check.

Reply:
oh, github burns me again by not flagging that a comment exists in between the one I see and the one I posted, even as I write a follow up

Reply:
Just for reference, it wasn't `dom-globals` that broke it as we haven't upgraded that. It was instead enabling strict mode for typescript.

Reply:
OHHH

thanks, that makes a lot more sense